### PR TITLE
Release a software package story

### PR DESCRIPTION
**Pull request by eccles**
Problem:
Example of release process using an SBOM.

Solution:
Added sbom entry to EVENTS_CREATE when releasing a software
package using an SBOM.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: archivist/sboms.py [hunk 1]
```diff
@@ -66,12 +69,149 @@ def __init__(self, archivist: "type_helper.Archivist"):
...
+
+        b = sbom["bom"]
+        m = b["metadata"]
+        c = m["component"]
+        hash_value = c["hashes"]["hash"]["#text"]
+        result = {
+            "author": c["author"],
+            "component": c["name"],
+            "hash": hash_value,
+            "repo": "",  # unsure about this
```
Comment (JAG-UK) [inline archivist/sboms.py:119]: repository is currently not reliable in SBOM specs. Suggest leaving it blank for now, can be patched later if someone comes to depend on it.

File: archivist/sboms.py [hunk 2]
```diff
@@ -66,12 +69,149 @@ def __init__(self, archivist: "type_helper.Archivist"):
...
+
+        # response to sbom upload contains all the info we need.
+        s = sbom.dict()
+        _, hash_value = s["hashes"][0].split(":")
+        result = {
+            "author": ",".join(s["authors"]),
+            "component": s["component"],
+            "identity": s["identity"],
+            "hash": hash_value,
+            "repo": "",  # unsure about this
```
Comment (JAG-UK) [inline archivist/sboms.py:201]: As prior comment, leave it blank for now.

File: functests/test_resources/sbom_story.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      - attributes:
+        - arc_display_name
+        - arc_namespace
+    behaviours:
+      - RecordEvidence
+      - Attachments
+    attributes:
+      arc_display_name: ACME Corporation Detector SAAS
+      arc_namespace: !ENV ${ARCHIVIST_NAMESPACE:namespace}
+      arc_display_type: SBOM
```
Comment (JAG-UK) [inline functests/test_resources/sbom_story.yaml:28]: We use "Software Package" as the arc_display_type in the other examples. This is more than just an SBOM - it's the complete release history of a package.

Comment (eccles) [reply functests/test_resources/sbom_story.yaml]: fixed - also the events now have a type of "Software Package Release" to be consistent

File: archivist/sboms.py (hunk 2 above)
Comment (eccles) [reply archivist/sboms.py]: removed the line

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**